python/desc/twinkles/validate_ic.py

```diff
@@ -0,0 +1,849 @@
+
+        df_sne = pd.read_csv(os.path.join(ic_folder, 'agn%s' % ic_file),
+                             delimiter=' ', header=None,
+                             names=base_columns+['internalExtinctionModel',
+                                                 'galacticExtinctionModel',
+                                                 'galacticAv', 'galacticRv'])
+
+        return df_galaxy, df_agn, df_sne
+
+    def process_sprinkled_agn(self, df_agn):
+
+        galtileids = []
+        twinkles_system = []
+        twinkles_im_num = []
+        galtile_list = np.genfromtxt(os.path.join(os.environ['TWINKLES_DIR'],
+                                                  'data', 'dc2_agn_cache.csv'),
+                                     delimiter=',', names=True, dtype=np.int)
+        
+        i=0
+        keep_idx = []
+        for agn_id in df_agn['uniqueId']:
+            twinkles_ids = np.right_shift(agn_id-117, 10)
+            galtileid = int(twinkles_ids/10000)
+            if galtileid in galtile_list['galtileid']:
+                keep_idx.append(i)
+                galtileids.append(galtileid)
+                twinkles_num = np.int(str(twinkles_ids)[-4:])
+                twinkles_system.append(twinkles_num // 4)
+                twinkles_im_num.append(twinkles_num % 4)
+                
+            i+=1
+
+        galtileids = np.array(galtileids, dtype=np.int)
+        sprinkled_agn = df_agn.iloc[np.array(keep_idx)]
+        sprinkled_agn = sprinkled_agn.reset_index(drop=True)
+        sprinkled_agn['galaxy_id'] = galtileids
+        sprinkled_agn['twinkles_system'] = twinkles_system
+        sprinkled_agn['image_number'] = twinkles_im_num
+        sprinkled_agn['lens_galaxy_uID'] = np.left_shift(galtileids, 10) + 97
+        sprinkled_agn['host_galaxy_bulge_uID'] = np.array(np.left_shift(galtileids*10000
+                                                    + np.array(twinkles_system, dtype=np.int)*4 +
+                                                    np.array(twinkles_im_num, dtype=np.int), 10) + 97, dtype=np.int)
+        sprinkled_agn['host_galaxy_disk_uID'] = np.array(np.left_shift(galtileids*10000
+                                                    + np.array(twinkles_system, dtype=np.int)*4 +
+                                                    np.array(twinkles_im_num, dtype=np.int), 10) + 107, dtype=np.int)
+            
+        return sprinkled_agn
+
+    def process_hosts(self, sprinkled_df, df_galaxy):
+
+        host_bulge_locs = []
+        host_disk_locs = []
+        for bulge_idx, disk_idx in zip(sprinkled_df['host_galaxy_bulge_uID'],
+                                       sprinkled_df['host_galaxy_disk_uID']):
+            bulge_matches = np.where(df_galaxy['uniqueId'] == bulge_idx)[0]
+            disk_matches = np.where(df_galaxy['uniqueId'] == disk_idx)[0]
+            for bulge_match, disk_match in zip(bulge_matches, disk_matches):
```

Comment:
Do you need to verify that `len(bulge_matches) == len(disk_matches)`?  `zip` will let you combine arrays of different lengths, truncating the longer of the two.